readthedocs/proxito/views/serve.py

```diff
@@ -483,41 +488,23 @@ def get(self, request, proxito_path, template_name='404.html'):
             version_slug=version_slug,
         )
 
-        if version_slug:
-            storage_root_path = final_project.get_storage_path(
-                type_="html",
-                version_slug=version_slug,
-                include_file=False,
-                version_type=self.version_type,
-            )
-
-            # First, check for dirhtml with slash
-            for tryfile in ("index.html", "README.html"):
-                storage_filename_path = build_media_storage.join(
-                    storage_root_path,
-                    f"{filename}/{tryfile}".lstrip("/"),
-                )
-                log.debug("Trying index filename.")
-                if build_media_storage.exists(storage_filename_path):
-                    log.info("Redirecting to index file.", tryfile=tryfile)
-                    # Use urlparse so that we maintain GET args in our redirect
-                    parts = urlparse(proxito_path)
-                    if tryfile == "README.html":
-                        new_path = parts.path.rstrip("/") + f"/{tryfile}"
-                    else:
-                        new_path = parts.path.rstrip("/") + "/"
-
-                    # `proxito_path` doesn't include query params.`
-                    query = urlparse(request.get_full_path()).query
-                    new_parts = parts._replace(
-                        path=new_path,
-                        query=query,
-                    )
-                    redirect_url = new_parts.geturl()
+        version = Version.objects.filter(
+            project=final_project, slug=version_slug
+        ).first()
 
-                    # TODO: decide if we need to check for infinite redirect here
-                    # (from URL == to URL)
-                    return HttpResponseRedirect(redirect_url)
+        # If we were able to resolve to a valid version, it means that the
+        # current file doesn't exist. So we check if we can redirect to its
+        # index file if it exists before doing anything else.
```

Comment:
What do you mean by "redirect ot its index file"? `/en/latest/file/` -> `/en/latest/file/index.html`?

If that's correct, can you expand the comment with an example similar to the one I wrote?